dll/win32/shell32/utils.h

```diff
@@ -45,3 +64,20 @@ SHELL32_SHExtractIcon(LPCWSTR File, int Index, int cx, int cy)
     int r = PrivateExtractIconsW(File, Index, cx, cy, &hIco, NULL, 1, 0);
     return r > 0 ? hIco : NULL;
 }
+
+HRESULT
+SHELL_CreateShell32DefaultExtractIcon(int IconIndex, REFIID riid, LPVOID *ppvOut);
+
+static inline HRESULT
+SHELL_CreateFallbackExtractIconForFolder(REFIID riid, LPVOID *ppvOut)
+{
+    const int id = IDI_SHELL_FOLDER;
+    return SHELL_CreateShell32DefaultExtractIcon(id > 1 ? -id : 0, riid, ppvOut);
```

Comment:
Yes, at compile time. Returning a negative (resource id) is better than index but we can't return -1 because things will break if somebody passes that to ExtractIcon where it has special meaning.